zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/AuthorizationCheckBehavior.java

```diff
@@ -452,18 +355,18 @@ public AuthorizedTenants getAuthorizedTenantIds(final TypedRecord<?> command) {
 
     final var username = getUsername(command);
     if (username.isPresent()) {
-      final var tenantIds =
-          new HashSet<>(
-              membershipState.getMemberships(EntityType.USER, username.get(), RelationType.TENANT));
-
-      final var groupIds =
-          membershipState.getMemberships(EntityType.USER, username.get(), RelationType.GROUP);
-      tenantIds.addAll(getTenantIdsForGroups(groupIds));
-
-      final var roleIds =
-          membershipState.getMemberships(EntityType.USER, username.get(), RelationType.ROLE);
-      tenantIds.addAll(getTenantIdsForRoles(roleIds));
+      final var tenantIds = getAuthorizedTenantIds(EntityType.USER, username.get()).toList();
+      if (tenantIds.isEmpty()) {
+        return AuthorizedTenants.DEFAULT_TENANTS;
```

Comment:
❓ This means that if my user doesn't have access to tenants (and multitenancy is enabled) I have access to the default tenants? This is different to earlier implementations where I only have implicit access to the default tenant if multi tenancy is disabled.

Reply:
How is this different to before? We had the same logic there, I didn't change it:
```
      if (tenantIds.isEmpty()) {
        return AuthorizedTenants.DEFAULT_TENANTS;
      } else {
        return new AuthenticatedAuthorizedTenants(tenantIds);
      }
```

Reply:
Maybe its something we can handle in a separate/focused PR?

Reply:
Yeah I guess we should just reopen https://github.com/camunda/camunda/issues/29518